validator-auction-rest-server/lib/EthEventsClient.js

```diff
@@ -106,28 +114,31 @@ export default class EthEventsClient {
         })
     }
 
-    getLowestBidPrice(allEvents) {
-        const filtered = allEvents.filter(e => e.event === 'AuctionEnded').map(e =>
-            new BN(e.args.find(a => a.name === 'lowestBidPrice')['value.hex'].substr(2), 16)
+    getLowestSlotPrice(allEvents) {
+        const filtered = allEvents.filter(e => e.event === 'AuctionEnded' || e.event === 'AuctionDepositPending').map(e =>
+            new BN(e.args.find(a => a.name === 'lowestSlotPrice')['value.hex'].substr(2), 16)
         )
         return filtered.length > 0 ? filtered[0] : undefined
     }
 
     getAuctionState(allEvents) {
-        const auctionStarted = allEvents.filter(e => e.event === 'AuctionStarted').length > 0
-        const auctionEnded = allEvents.filter(e => e.event === 'AuctionEnded').length > 0
-        const auctionFailed = allEvents.filter(e => e.event === 'AuctionFailed').length > 0
-        const auctionDeployed = allEvents.filter(e => e.event === 'AuctionDeployed').length > 0
-        if (auctionEnded) {
-            return 'Finished'
-        } else if (auctionFailed) {
-            return 'Failed'
-        } else if (auctionStarted) {
-            return 'Started'
-        } else if (auctionDeployed) {
-            return 'Not Started'
+        const started = allEvents.filter(e => e.event === 'AuctionStarted').length > 0
+        const ended = allEvents.filter(e => e.event === 'AuctionEnded').length > 0
+        const failed = allEvents.filter(e => e.event === 'AuctionFailed').length > 0
+        const deployed = allEvents.filter(e => e.event === 'AuctionDeployed').length > 0
+        const depositPending = allEvents.filter(e => e.event === 'AuctionDepositPending').length > 0
+        if (ended) {
+            return STATE_FINISHED
+        } else if (failed) {
+            return STATE_FAILED
+        } else if (started) {
+            return STATE_STARTED
+        } else if (deployed) {
+            return STATE_NOT_STARTED
+        } else if (depositPending) {
```

Comment:
This is never reachable I think, because a deposit pending auction is always also started. 
I guess this has to be after failed.

Reply:
Very good point!